clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2276,6 +2415,11 @@ private Node leader(PartitionInfo partitionInfo) {
         return partitionInfo.leader();
     }
 
+    // This is used in the describe topics path if using DescribeTopics API.
+    private Node replicaToFakeNode(int id) {
+        return new Node(id, "Dummy", 0);
+    }
```

Comment:
Just for my own understanding, why do we favor creating _fake_ nodes instead of looking up the _real_ nodes from the metadata or something?

Reply:
The DescribeTopicParitions does not provide the node info as Metadata Api does. However the TopicPartitionInfo constructor requires the node info, but the node info is useless in the describeTopic scenario.

Reply:
Maybe you should create a new constructor for TopicPartitionInfo that doesn't require a Node. This faking is a bit ugly.

Reply:
Now it will query the cluster info for the nodes.